flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/AdaptiveScheduler.java

```diff
@@ -1048,8 +1044,8 @@ public void goToExecuting(
                         this,
                         userCodeClassLoader,
                         failureCollection,
-                        settings.getScalingIntervalMin(),
-                        settings.getScalingIntervalMax()));
+                        DefaultRescaleManager.Factory.fromSettings(settings),
```

Comment:
Could  we create the `DefaultRescaleManager.Factory` inside of AdaptiveScheduler constructor?

I think the `DefaultRescaleManager.Factory` can be reuse, and we can create different DefaultRescaleManager using same factory. WDYT?

Reply:
sure, good point. That's possible. 👍